standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -836,28 +835,24 @@ public void run() {
 
          LeaderElectionContext context = new LeaderElectionContext.ContextBuilder(conf)
              .setHMSHandler(thriftServer.getHandler()).servHost(getServerHostName())
-             // always tasks
-             .setTType(LeaderElectionContext.TTYPE.ALWAYS_TASKS)
+             .setTType(LeaderElectionContext.TTYPE.ALWAYS_TASKS) // always tasks
              .addListener(new HouseKeepingTasks(conf, false))
-             // housekeeping tasks
-             .setTType(LeaderElectionContext.TTYPE.HOUSEKEEPING)
+             .setTType(LeaderElectionContext.TTYPE.HOUSEKEEPING) // housekeeping tasks
              .addListener(new CMClearer(conf))
              .addListener(new StatsUpdaterTask(conf))
              .addListener(new CompactorTasks(conf, false))
              .addListener(new CompactorPMF())
              .addListener(new HouseKeepingTasks(conf, true))
-             // compactor worker
-             .setTType(LeaderElectionContext.TTYPE.WORKER)
-             .addListener(new CompactorTasks(conf, true), MetastoreConf.getVar(conf,
-                 MetastoreConf.ConfVars.HIVE_METASTORE_RUNWORKER_IN).equals("metastore"))
+             .setTType(LeaderElectionContext.TTYPE.WORKER) // compactor worker
+             .addListener(new CompactorTasks(conf, true),
+                 MetastoreConf.getVar(conf, MetastoreConf.ConfVars.HIVE_METASTORE_RUNWORKER_IN).equals("metastore"))
              .build();
           if (shutdownHookMgr != null) {
             shutdownHookMgr.addShutdownHook(() -> context.close(), 0);
           }
           context.start();
         } catch (Throwable e) {
-          LOG.error("Failure when starting the leader tasks, compactions or housekeeping tasks may not happen, " +
-              StringUtils.stringifyException(e));
+          LOG.error("Compaction or Housekeeping tasks may not happen", e);
```

Comment:
Should we keep the previous logging message L#859? I think it gives more details than the current suggestion.